File: 2015/Day1/Day1.1.py
import os

if __name__ == "__main__":
    with open("input.txt", "r") as f:
        string = f.readline()

    floor = 0
    for c in string:
        if c == "(":
            floor += 1
        elif c == ")":
            floor -= 1

    print(floor)

File: 2015/Day1/Day1.1.c
#include <stdio.h>

int main() {
  char c;
  int floor = 0;

  char in[65536];
  FILE *fp = fopen("input.txt", "r");
  fgets(in, sizeof(in), fp);

  int i = 0;
  while ((c = in[i++]) != '\n' && c != EOF) {
    if (c == '(') {
      floor++;
    } else if (c == ')') {
      floor--;
    }
  }

  printf("%d", floor);
}